### PR TITLE
Fix #12023, accidental disabling of the GL backend on non-Windows ARM platforms caused by #12013.

### LINKED ISSUE #12023: PPSSPP v1.8.0-180-g5fe02a4ea unable to initialize rendering engine
### SCREENSHOTS
Screenshot_2019-05-08-08-32-02

### What happens?

I can't play any games in the latest ppsspp buildbot
### What *should* happen?
I can play without problem in ppsspp v1.8.0-155

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.
Android 5.1 Lollipop
Mali-450 MP GPU
Octacore CPU

### PR DESCRIPTION
**Pull request by hrydgard**

### CODE REVIEW

**Comment by unknownbrackets** [comment]
If this means _M_ARM is defined there, don't we need to do this in Core/Config.cpp, UI/EmuScreen.cpp, UI/GameSettingsScreen.cpp, and more places in gpu_features.cpp?

-[Unknown]

**Comment by hrydgard** [comment]
Right.. I'll spend a bit more time on it tonight, cleaning up the defines so we use consistent architecture defines across the OS:es.

**Comment by hrydgard** [comment]
Updated with a whole host of additional changes. Tried to fix the ARM/ARM64 desktop debug builds, almost there but some errors remain, like these:

```

2>ARM\SPIRV-CrossDebug\spirv_cfg.obj : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'ARM'
2>Done building project "SPIRV-Cross.vcxproj" -- FAILED.
1>ARM\Debug\CodeGen.obj : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'ARM'
1>Done building project "glslang.vcxproj" -- FAILED.
3>_Output\libarmips\DebugARM\Architecture.obj : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'ARM'
3>Done building project "libarmips.vcxproj" -- FAILED.
4>ARM\Debug\connection_win.obj : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'ARM'
```
Really not sure where those are from. But the release builds build and the issue is fixed.

**Comment by hrydgard** [comment]
Ugh, now no error message but graphics is broken on Android ARM32... wtf

**Comment by hrydgard** [comment]
Ah, doh. Fixed.

File: Common/Common.vcxproj [hunk 1]
```diff
@@ -38,7 +38,7 @@
     <ProjectGuid>{3FCDBAE2-5103-4350-9A8E-848CE9C73195}</ProjectGuid>
     <Keyword>Win32Proj</Keyword>
     <RootNamespace>Common</RootNamespace>
-    <WindowsTargetPlatformVersion></WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
```
Comment (unknownbrackets) [inline Common/Common.vcxproj]: Does this intentionally revert 05649452?

-[Unknown]

File: Core/Config.cpp [hunk 2]
```diff
@@ -566,12 +566,12 @@ int Config::NextValidBackend() {
...
 		}
 #endif
-#if !PPSSPP_PLATFORM(UWP) && !defined(_M_ARM64) && !defined(_M_ARM)
+#if PPSSPP_API(ANY_GL)
 		if (!failed.count((int)GPUBackend::OPENGL)) {
 			return (int)GPUBackend::OPENGL;
 		}
 #endif
-#if PPSSPP_PLATFORM(WINDOWS) && !PPSSPP_PLATFORM(UWP)
+#if PPSSPP_PLATFORM(D3D9)
```
Comment (unknownbrackets) [inline Core/Config.cpp]: *PPSSPP_API

-[Unknown]

File: headless/Headless.cpp [hunk 3]
```diff
@@ -277,9 +277,13 @@ int main(int argc, const char* argv[])
...
 		// Default to GLES if no value selected.
-		else if (!strcmp(argv[i], "--graphics"))
+		else if (!strcmp(argv[i], "--graphics")) {
+#if PPSSPP_API(ANY_GL)
 			gpuCore = GPUCORE_GLES;
-		else if (!strncmp(argv[i], "--screenshot=", strlen("--screenshot=")) && strlen(argv[i]) > strlen("--screenshot="))
+#else
+			gpuCore = GPUCORE_DIRECTX11;
+#endif
+		}  else if (!strncmp(argv[i], "--screenshot=", strlen("--screenshot=")) && strlen(argv[i]) > strlen("--screenshot="))
```
Comment (unknownbrackets) [inline headless/Headless.cpp]: nit: Two spaces before `else`.

-[Unknown]

**Review by unknownbrackets — APPROVED** [review]
Looks like this might break the D3D9 fallback, though it's probably a better problem than the GL breakage.

-[Unknown]

File: Common/Common.vcxproj (hunk 1 above)
Comment (unknownbrackets) [reply Common/Common.vcxproj]: Nevermind, failed to read the commit message.... sorry.

-[Unknown]

**Comment by driver1998** [comment]
> 2>ARM\SPIRV-CrossDebug\spirv_cfg.obj : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'ARM'
> 2>Done building project "SPIRV-Cross.vcxproj" -- FAILED.
> 1>ARM\Debug\CodeGen.obj : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'ARM'
> 1>Done building project "glslang.vcxproj" -- FAILED.
> 3>_Output\libarmips\DebugARM\Architecture.obj : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'ARM'
> 3>Done building project "libarmips.vcxproj" -- FAILED.
> 4>ARM\Debug\connection_win.obj : fatal error

Looks like something is hard coded to compile as x86... I got some similar errors before. Let me check.

**Comment by driver1998** [comment]
So Debug ARM builds for me, not sure if that is fixed on you as well.

File: GPU/GPU.cpp [hunk 4]
```diff
@@ -39,8 +36,6 @@
 #include "GPU/D3D11/GPU_D3D11.h"
```
Comment (driver1998) [inline GPU/GPU.cpp]: Add a PPSSPP_API(D3D9) detection here maybe? It seems to break UWP compilation for now.

Comment (driver1998) [reply GPU/GPU.cpp]: Well I am talking about the line before it
the DirectX9 inlcude

Comment (hrydgard) [reply GPU/GPU.cpp]: Done.

**Pull request merged**